libs/core/src/lib/button/button.component.ts

```diff
@@ -21,17 +20,26 @@ export type ButtonOptions = 'light' | 'emphasized' | 'menu';
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ButtonComponent extends AbstractFdNgxClass {
+export class ButtonComponent implements OnInit, CssClassBuilder, CssStyleBuilder {
+    private _class: string = '';
+    @Input() set class(userClass: string) {
+        this._class = userClass;
+        this.buildComponentCssClass();
+    } // user's custom classes
+
+    /** The icon to include in the button. See the icon page for the list of icons. */
+    private _glyph: string
+    @Input() set glyph(icon: string) {
+        this._glyph = icon;
+        this.buildComponentCssClass()
+    };
```

Comment:
This is great, but I'm not sure what about situation, when someone changes for ex. `compact` property or `fdType`. Will the new classes be applied on this element?

Reply:
in that case, seter is needed or ngonchange although it is not recomended since every change whole css style will be rebuilded